proof-producer/libs/assigner/src/utils.cpp

```diff
@@ -0,0 +1,110 @@
+#include "zkevm_framework/assigner_runner/utils.hpp"
+
+#include <evmc.h>
+
+#include <cstdint>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+#include "zkevm_framework/core/types/message.hpp"
+
+evmc_call_kind evmc_msg_kind(const std::bitset<8>& type) {
+    if (type.test(std::size_t(core::types::MessageKind::Deploy))) {
+        return EVMC_CREATE;
+    } else if (type.test(std::size_t(core::types::MessageKind::Internal))) {
+        return EVMC_CALL;
+    } else {
+        return EVMC_CALL;
+    }
+}
+
+evmc_address to_evmc_address(const core::types::Address& v) {
+    evmc_address res;
+    for (std::size_t i = 0; i < core::types::ADDR_SIZE; i++) {
+        res.bytes[i] = std::to_integer<uint8_t>(v[i]);
+    }
+    return res;
+}
+
+template<typename T>
+static std::string byte_array_to_str(const T* data, size_t size) {
+    std::stringstream os;
+    os << std::hex << "0x";
+    for (int i = 0; i < size; ++i) {
+        os << std::setw(2) << std::setfill('0') << (int)data[i];
+    }
+    return os.str();
+}
+
+std::string to_str(const evmc_address& v) { return byte_array_to_str(v.bytes, sizeof(v.bytes)); }
+
+std::string to_str(const core::types::Address& v) { return byte_array_to_str(v.data(), v.size()); }
+
+std::string to_str(const evmc_bytes32& v) { return byte_array_to_str(v.bytes, sizeof(v.bytes)); }
+
+std::string to_str(const core::Hash& v) { return byte_array_to_str(v.data(), v.size()); }
+
+std::string to_str(const std::vector<uint8_t>& code) {
+    return byte_array_to_str(code.data(), code.size());
+}
+
+std::string to_str(const std::bitset<8>& type) {
+    if (type.test(std::size_t(core::types::MessageKind::Deploy))) {
+        return "EVMC_CREATE";
+    } else if (type.test(std::size_t(core::types::MessageKind::Internal))) {
+        return "EVMC_CALL";
+    } else {
+        return "unknown message type";
+    }
+}
+
+std::string to_str(evmc_status_code code) {
```

Comment:
I'm not sure, but maybe it's better to override operator<< insread of creating functuons to_str and similar?
Most probably you want to print these to some output, it's easier to write LOG << code << ...